webdataset/writer.py

```diff
@@ -316,10 +316,8 @@ def __init__(
             else:
                 tarmode = "w|gz" if fileobj.endswith("gz") else "w|"
             fileobj = gopen(fileobj, "wb")
-            self.own_fileobj = fileobj
         else:
             tarmode = "w|gz" if compress is True else "w|"
-            self.own_fileobj = None
```

Comment:
My problem arrises here, when the opened fileobj is not saved anywhere, so I won't have any reference to the opened file and python seems to do nothing when the object is garbage collected.
Maybe I have misunderstood the concept of opener,
mine is defined 
```python
def cloudpathlib_opener(filename, flags):
    return AnyPath(filename).open(mode=flags)
```